### ISSUE TITLE
ddtrace incompatbility ?

### ISSUE THREAD

**Issue by gingerlime** [DataDog/dogstatsd-ruby#197]
Seeing this warning after trying to upgrade ...

```
WARN -- ddtrace: [ddtrace] This version of `ddtrace` is incompatible with `dogstastd-ruby` version >= 5.0 and can cause unbounded memory usage. Please use `dogstastd-ruby` version < 5.0 instead.
```

**Comment by remeh** [comment]
Hey @gingerlime,
We are working on this to restore the compatibility. To be complete, it works but it can lead to unbounded memory usage in some use-cases. This warning has been added to ensure that people know about it, and meanwhile, to be sure that nobody has any issues, we recommend to pin the dogstatsd-ruby gem using: `gem 'dogstatsd-ruby', '~> 4.0'` which is known to work properly.

**Comment by gingerlime** [comment]
Thank you @remeh

**Comment by remeh** [comment]
Closing this since this is fixed using `dogstatsd-ruby 5.2.0` and `dd-trace-rb 0.51.0` and the warning would not appear anymore.
